### PR TITLE
Basic functionality

### PR DESCRIPTION
**Pull request by David-Lanzz**
- I have successfully added reducer actions for adding and removing books from the store/Dom
- No linter errors
- Correct use of gitflow

### CODE REVIEW

File: src/redux/features/books/bookSlice.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+};
+const bookSlice = createSlice({
+  name: 'books',
+  initialState,
+  reducers: {
+    removeBooks: (state, action) => {
+      const id = action.payload;
+      const filtered = state.books.filter((book) => book.item_id !== id);
+      return { ...state, books: filtered };
+    },
```
Comment (Qoosim) [inline src/redux/features/books/bookSlice.js:31]: - [ ] Great job so far. However, there's littile issue here. I was able to delete all the populated book objects without any issue. But after adding new books, I was unable to remove them one after the other, but remove all the added books at once when intending to remove only just one book. Kindly look at it and fix.

### Your App Behavior
[app-behave.webm](https://user-images.githubusercontent.com/26597432/230059307-7afd0f89-1be3-4cf2-916e-deffe422f9f0.webm)

- You may consider checking this approach :point_down: 
```
  removeBook: (state, action) => {
      state.books = state.books.filter((book) => book.id !== action.id)
    },

```

File: src/redux/features/categories/categorySlice.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+const initialState = {
+  categories: [],
+};
+
+const categorySlice = createSlice({
+  name: 'categories',
+  initialState,
+});
+export default categorySlice.reducer;
```
Comment (Qoosim) [inline src/redux/features/categories/categorySlice.js]: - [ ] It is required that when `check status` button is clicked, it displays  `Under construction` text. At this moment, that requirement has not been implemented. Kindly attend to it.

[app-2.webm](https://user-images.githubusercontent.com/26597432/230061317-0696ae1c-b320-4c1b-92a6-47b644b2fc0c.webm)

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle:

Hi @David-Lanzz, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Git Flow. :heavy_check_mark: 




## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by OyePriscilla — APPROVED** [review]
Hi @David-Lanzz 👏 👏 👏 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
Use store, actions and reducers in React ✔️ 
Connect React and Redux ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**